### PR TITLE
User Model functions

### PR DESCRIPTION
**Pull request by Deon**
`User` and `Friend` models.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "typescript.tsdk": "./node_modules/typescript/lib"
```
Comment (icechen1) [inline .vscode/settings.json]: jokes thanks VS Code

File: test/models.js [hunk 2]
```diff
@@ -1,17 +1,104 @@
...
+
+    it('should find first user by query', (done) => {
+      User.first(['id', 'username', 'email'], {id : 1}).then((user) => {
+        assert.typeOf(user, 'object');
+        assert.equal(user.id, 1);
+        done();
+      });
+    });
+
+    after(() => {
```
Comment (Deon) [inline test/models.js]: This might be clearer if we leave it at the top (that won't change behaviour).

File: test/models.js [hunk 3]
```diff
@@ -1,17 +1,104 @@
...
+          assert.typeOf(user, 'object');
+          assert.notEqual(user.password, "test2");
+          // check for bcrypt password header
+          assert.equal(user.password.slice(0,4), "$2a$")
+          done();
+        });
+      });
+    });
+
+    after(() => {
```
Comment (Deon) [inline test/models.js]: If this is the same for all tests, put it at the describe level, in an `afterEach`?

**Comment by Deon** [comment]
Resolves #5

**Pull request merged**